### DISCUSSION TITLE
StaticFiles and Session Cookies

### DISCUSSION THREAD

**Discussion by lrpatterson** [fastapi/fastapi#6530] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()

app.mount("/submission", StaticFiles(directory=settings.static_file_location, html=True), name="static")

origins = [
    "*"
]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["OPTIONS", "GET", "POST"],
    allow_headers=[],
)


app.add_middleware(SessionMiddleware, secret_key='secret')
```

### Description

So I have a bundled js/html site that is served via `app.mount`. All that works fine, but I am wanting to also to set a session variable (captcha data) for the user when they load the site from the '/submissions' endpoint. Is this currently possible (ie capture the incoming request and set a session variable) via StaticFiles.

### Operating System

Windows

### FastAPI Version

068.0

### Python Version

Python 3.8.6rc1

**Comment by lrpatterson** [comment, marked as answer]
I figured out that I just needed to serve my bundled site via a Jinja template.  That allows me to create a session upon a users visit.

**Reply by xpatronum** [reply]
Could u elaborate more, plz ?

What do you mean by JINJA templates, do we need to serve `index.html` as JINJA template or what do u mean, eloborate plz?
